frontend/src/main/scala/bloop/engine/Interpreter.scala

```diff
@@ -280,6 +279,26 @@ object Interpreter {
     }
   }
 
+  private def missingScalaNativeToolchain(state: State, project: Project): Task[State] = {
+    val artifactName = project.platform match {
+      case Config.Platform.Native(config) => ScalaNativeToolchain.artifactNameFrom(config.version)
+      case _ => throw new IllegalArgumentException
+    }
+
+    val msg = s"Artifact $artifactName for Scala Native toolchain could not be resolved in project '$project'"
+    state.withError(msg, ExitStatus.LinkingError)
+  }
+
+  private def missingScalaJsToolchain(state: State, project: Project): Task[State] = {
+    val artifactName = project.platform match {
+      case Config.Platform.Js(config) => ScalaJsToolchain.artifactNameFrom(config.version)
+      case _ => throw new IllegalArgumentException
```

Comment:
Let's remove these `IllegalArgumentException`. If we hit this path, the project isn't directly a javascript project so we'd better report that explicitly to users.

Reply:
Should I write `sys.error("Scala.js project expected")` instead?

Reply:
What about "Fatal error: Scala.js project ${project.name} doesn't have a js config. Report upstream."

Reply:
Sounds good, but I will change "js config" to "JavaScript configuration".